modules/imap/handler_modules.php

```diff
@@ -685,6 +685,82 @@ public function process() {
     }
 }
 
+
+/**
+ * Archive a message
+ * @subpackage imap/handler
+ */
+class Hm_Handler_imap_archive_message extends Hm_Handler_Module {
+    /**
+     * Use IMAP to archive the selected message uid
+     */
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_msg_uid', 'imap_server_id', 'folder'));
+
+        if ($success) {
+            $cache = Hm_IMAP_List::get_cache($this->cache, $form['imap_server_id']);
+            $imap = Hm_IMAP_List::connect($form['imap_server_id'], $cache);
+            $archive_folder = false;
+
+            $specials = $this->user_config->get('special_imap_folders', array());
+            if (array_key_exists($form['imap_server_id'], $specials)) {
+                if (array_key_exists('archive', $specials[$form['imap_server_id']])) {
+                    if ($specials[$form['imap_server_id']]['archive']) {
+                        $archive_folder = $specials[$form['imap_server_id']]['archive'];
+                    }
+                }
+            }
+
+            if(!$archive_folder) {
+                $archive_folder = "Archive";
+            }
+
+            if (imap_authed($imap)) {
+                $new_folder = $archive_folder . '/' . hex2bin($form['folder']);
```

Comment:
ok, $imap->delim did NOT fix the issue :) it must be set to false which caused the 2 folders to be created in my case. I will dig more into it.

Reply:
pretty sure we need to call $imap->get_namespaces() to get the delimiter, and then this will be good to go. I will let follow up as soon as I can.

Reply:
I made update to use delimiter used by the imap server

Reply:
Looks good! I am going to merge this as is then make a few changes, specifically:
- we have a function to properly format a folder for create_mailbox(), however it lives in the imap_folders module so I'm going to move it to the imap module and use it
- when using caching new folders are not showing up without logging out and back in. This is unrelated to your changes and I just fixed it in master for the imap_folders module set, so I will apply that same fix here.
- the logic that creates the folders then moves the message is a bit hard to follow, I'm going to do some refactoring there to make it easier to understand but it won't change the functionality.
- Have you tried archiving a message in a subfolder, that is in another subfolder? like FOO/BAR/BAZ? I'm curious what will happen there. I think it might not work but have not tested it yet.

Overall great work and thanks for doing this!